ceph/ceph.go

```diff
@@ -304,22 +429,28 @@ func createNamespace(ns string) core.Namespace {
 
 // getCephDaemonNames scans the path to ceph sockets in search of an instance which name contains specified prefix and extension.
 // Returns names of available ceph-daemon sockets
-func (s *Socket) getCephDaemonNames() []string {
+func (s *Socket) getCephDaemons() ([]Daemon, error) {
 	files, err := ioutil.ReadDir(s.path)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Not found any ceph-daemon socket in %+v. Check if the given path is correct\n", s.path)
-		panic(err)
+		return nil, err
 	}
-	names := []string{}
+	daemons := []Daemon{}
 	suffix := "." + s.ext
 	for _, f := range files {
 		// read names of all items in socket path with specific prefix-name (defaults to "ceph-") and extension (defaults to .asok)
 		if strings.HasPrefix(f.Name(), s.prefix) && strings.HasSuffix(f.Name(), suffix) {
-			names = append(names, f.Name())
+			daemonNameSplit := strings.Split(trimPrefixAndSuffix(f.Name(), s.prefix, "."+s.ext), ".")
+			daemonName := daemonNameSplit[0]
+			daemonID := ""
+			if len(daemonNameSplit) > 1 {
+				daemonID = daemonNameSplit[1]
+			}
+			daemons = append(daemons, Daemon{id: daemonID, name: daemonName, fullName: f.Name()})
```

Comment:
There are always IDs associated with daemons, so this check can be removed at all, I added this because I didn't know that for sure.